[CI] run-partner-cuj.sh exits 1 even when all tests pass — 30일 연속 daily-backend-simulation 실패

## Summary

`daily-backend-simulation` 워크플로우가 **최근 30회 연속** 실패 중. 원인 조사 결과 **테스트는 모두 통과하는데 shell script가 exit 1 반환**. 인프라/테스트 문제 아님.

## Evidence

- 최신 실행 [`24806377684`](https://github.com/Mark-Yun/minglit/actions/runs/24806377684) 의 `partner-cuj-test` 로그:
  - `cuj_application_action`: 7/7 ✅
  - `cuj_application_review`: 7/7 ✅
  - `cuj_checkin_manage`: 4/4 ✅
  - `cuj_checkin_qr`: 9/9 ✅
  - `cuj_event_create_wizard`: 19/19 ✅
  - `cuj_onboarding_to_event`: 8/8 ✅
  - `cuj_partner_account_deletion`: 8/8 ✅
  - `cuj_recurring_event`: 17/17 ✅
  - `cuj_settlement`: 6/6 ✅
  - `cuj_team_management`: 8/8 ✅
  - `partner_redirect`: 10/10 ✅
  - **총 103 tests passed, 0 failed**
- 스크린샷 41개 업로드 성공 (1.1MB)
- 직후 로그: `##[error]The process '/usr/bin/sh' failed with exit code 1`
- `client-cuj-test`는 항상 success (또는 partner 실패로 인한 cancelled) — 동일 인프라, 다른 스크립트만이 차이

## Root Cause

`.github/scripts/run-partner-cuj.sh` 마지막 줄:

```bash
[ "$found" -eq 0 ] && echo "⏭ No integration tests found, skipping."
```

1. 테스트 발견 시 `found=1`
2. 종료 직전 `[ 1 -eq 0 ]` 평가 → **exit status 1**
3. `&&` short-circuit으로 echo는 실행 안 됨
4. 이게 스크립트의 **마지막 명령** → 스크립트 종료 코드 = 1
5. `set -euo pipefail` 때문에 그대로 전파

`run-client-cuj.sh`는 다른 패턴이라 안 걸림:
```bash
for f in test/integration/*_test.dart; do
  [ -f "$f" ] || { echo "⏭ ..."; break; }
  flutter test ...  # 마지막 명령 → 종료 코드 0
done
```

## Fix

`.github/scripts/run-partner-cuj.sh` 마지막 줄을 교체:

```bash
# Before
[ "$found" -eq 0 ] && echo "⏭ No integration tests found, skipping."

# After
if [ "$found" -eq 0 ]; then
  echo "⏭ No integration tests found, skipping."
fi
```

또는 `|| true` suffix 허용.

## Impact

- 30일 연속 daily CUJ 회귀 감지 공백 — partner 쪽 103 테스트가 CI에서 깨지더라도 눈치 못 챘을 수 있음
- Screenshot 아티팩트는 업로드되지만 workflow failure 때문에 사용자 알림이 CI-failure로 묶여 신호가 섞임

## Verification

PR merge 후 다음 daily 실행 (22:xx UTC)에서 `partner-cuj-test` 성공 확인 → 30일 체인 해소.

--- a/.github/scripts/run-partner-cuj.sh
+++ b/.github/scripts/run-partner-cuj.sh
@@ -9,4 +9,5 @@ for f in test/integration/*_test.dart; do
   echo "▶ Running: $f"
   flutter test "$f" --flavor dev --dart-define-from-file=../../minglit_env/dev/flutter.env
 done
-[ "$found" -eq 0 ] && echo "⏭ No integration tests found, skipping."
+# Fix #1755: [ ... ] && echo 패턴은 found=1 일 때 마지막 명령이 exit 1 → 스크립트 종료 코드 1
+if [ "$found" -eq 0 ]; then echo "⏭ No integration tests found, skipping."; fi

